go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -705,3 +705,11 @@ func (bnc *BaseNetworkController) recordNodeErrorEvent(node *kapi.Node, nodeErr
 	klog.V(5).Infof("Posting %s event for Node %s: %v", kapi.EventTypeWarning, node.Name, nodeErr)
 	bnc.recorder.Eventf(nodeRef, kapi.EventTypeWarning, "ErrorReconcilingNode", nodeErr.Error())
 }
+
+func (bnc *BaseNetworkController) doesNetworkTolerateIPAMLessConfig() bool {
+	return bnc.TopologyType() == types.Layer2Topology && len(bnc.Subnets()) == 0
+}
+
+func (bnc *BaseNetworkController) doesNetworkRequireIPAM() bool {
+	return !bnc.doesNetworkTolerateIPAMLessConfig()
+}
```

Comment:
I have nothing against added expressiveness with additional variables/methods.

But in this specific case, it was not clear cut to me that `doesNetworkTolerateIPAMLessConfig` is easier to the eyes than `!doesNetworkRequireIPAM`

This is highly subjective though and not asking to change anything, just giving you an outside view on it.

Reply:
Also, I have the impression that `Tolerate` is not the correct wording. It either has IPAM or it has not. `Tolerate` kinda gives the impression that the final decision is taken elsewhere with additional criteria and that this is just one condition of many.